app/models/post.rb

```diff
@@ -0,0 +1,19 @@
+class Post < ApplicationRecord
+  belongs_to :user
+  has_many :comments
+  has_many :likes
+
+  after_create :increment_posts_counter
+  before_destroy :decrement_posts_counter
+  def increment_posts_counter
+    user.increment!(:posts_counter)
+  end
+
+  def decrement_posts_counter
+    user.decrement!(:posts_counter)
+  end
```

Comment:
- [ ] Kindly change `user` to `author` to match the changes we have made to the associations. 

## This applies to the Comment model as well